test/e2e/extensions/high_availability_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extensions
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"agones.dev/agones/pkg/util/runtime"
+	e2eframework "agones.dev/agones/test/e2e/framework"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// Test creating a gameserver when one of the extensions pods is down/deleted
+func TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown(t *testing.T) {
+	logger := e2eframework.TestLogger(t)
+	ctx := context.Background()
+
+	if !runtime.FeatureEnabled(runtime.FeatureSplitControllerAndExtensions) {
+		logger.Infof("Skip test. SplitControllerAndExtensions feature is not enabled: flag is %v", runtime.FeatureEnabled(runtime.FeatureSplitControllerAndExtensions))
+		return
+	}
+
+	assert.NoError(t, waitForAgonesExtensionsRunning(ctx))
+
+	list, err := getAgoneseExtensionsPods(ctx)
+	logger.Infof("Length of pod list is %v", len(list.Items))
+	if err != nil {
+		t.Fatalf("Could not get list of Extension pods: %v", err)
+	}
+	if len(list.Items) <= 1 {
+		t.Fatal("Cluster has no Extensions pod or has only 1 extensions pod")
+	}
+
+	logger.Info("Removing one of the Extensions Pods")
+	err = deleteAgonesExtensionsPods(ctx)
+	assert.NoError(t, err)
+
+	assert.NoError(t, waitForAgonesExtensionsRunning(ctx))
+
+	logger.Info("Creating default game server")
+	gs := framework.DefaultGameServer(defaultNs)
+	readyGs, err := framework.CreateGameServerAndWaitUntilReady(t, defaultNs, gs)
+	if err != nil {
+		t.Fatalf("Could not get a GameServer ready: %v", err)
+	}
+	logger.WithField("gsKey", readyGs.ObjectMeta.Name).Info("GameServer Ready")
+
+	gsClient := framework.AgonesClient.AgonesV1().GameServers(defaultNs)
+	defer gsClient.Delete(ctx, readyGs.ObjectMeta.Name, metav1.DeleteOptions{}) // nolint: errcheck
+}
+
+// deleteAgonesExtensionsPod deletes one of the extensions pod for the Agones extensions,
+// faking a extensions pod crash.
+func deleteAgonesExtensionsPods(ctx context.Context) error {
+	list, err := getAgoneseExtensionsPods(ctx)
+	if err != nil {
```

Comment:
here and below, I'd be inclined to throw in a `t *testing.T` parameter and just use `assert.NoError(t, err)`. Keeps it more concise and better tracks the line number of failures.

Reply:
Done! As for the method below `waitForAgonesExtensionsRunning`, I don't think it makes sense to add `t *testing.T` as it won't be able to use it since it still needs to return the bool.